src/Component/FieldTypes/CallbackFieldType.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Grid\FieldTypes;
+
+use Sylius\Component\Grid\DataExtractor\DataExtractorInterface;
+use Sylius\Component\Grid\Definition\Field;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class CallbackFieldType implements FieldTypeInterface
+{
+    private DataExtractorInterface $dataExtractor;
+
+    public function __construct(DataExtractorInterface $dataExtractor)
+    {
+        $this->dataExtractor = $dataExtractor;
+    }
+
+    public function render(Field $field, $data, array $options): string
+    {
+        $value = $this->dataExtractor->get($field, $data);
+        $value = call_user_func($options['callback'], $value);
+
+        if ($options['htmlspecialchars'] !== true) {
+            return (string) $value;
+        }
```

Comment:
Just a suggestion, looks a bit unusual to have `!== true`on bool typed property.

```suggestion
        if ($options['htmlspecialchars'] === false) {
            return (string) $value;
        }
```
```suggestion
        if (!$options['htmlspecialchars']) {
            return (string) $value;
        }
```